src/wasm/wasm-validator.cpp

```diff
@@ -1960,7 +1960,7 @@ void FunctionValidator::visitMemoryGrow(MemoryGrow* curr) {
 }
 
 void FunctionValidator::visitRefNull(RefNull* curr) {
-  shouldBeTrue(getModule()->features.hasReferenceTypes(),
+  shouldBeTrue(!getFunction() || getModule()->features.hasReferenceTypes(),
```

Comment:
```suggestion
  // If we are not in a function, this is a global location like a table. We allow
  // RefNull there as we represent tables that way regardless of what features
  // are enabled.
  shouldBeTrue(!getFunction() || getModule()->features.hasReferenceTypes(),
```
and similar down below

Reply:
Done.